pkg/controller/controlplane/controlplane_controller_test.go

```diff
@@ -110,3 +129,460 @@ func TestReconcile(t *testing.T) {
 	g.Expect(err).NotTo(gomega.HaveOccurred())
 	g.Eventually(requests, timeout).Should(gomega.Receive(gomega.Equal(expectedRequest)))
 }
+
+func TestReconcileFail(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	expectedRequest := reconcile.Request{NamespacedName: types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"}}
+	cpKey := types.NamespacedName{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace"}
+
+	cluster := &clusterv1alpha1.EKS{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-eks-name", Namespace: "test-controlplane-namespace"},
+		Spec: clusterv1alpha1.EKSSpec{
+			AccountID: "test-account-id",
+			ControlPlane: clusterv1alpha1.ControlPlaneSpec{
+				ClusterName: "test-cluster-name",
+			},
+			CrossAccountRoleName: "test-crossaccount-role",
+			NodeGroups:           []clusterv1alpha1.NodeGroupSpec{{Name: "test-nodegroup-name", IAMPolicies: []clusterv1alpha1.Policy{}}},
+			Region:               "test-region-id",
+		},
+	}
+
+	instance := &clusterv1alpha1.ControlPlane{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-controlplane-name", Namespace: "test-controlplane-namespace", Labels: map[string]string{"eks.owner.name": "test-eks-name", "eks.owner.namespace": "test-controlplane-namespace"}},
+		Spec: clusterv1alpha1.ControlPlaneSpec{
+			ClusterName: "test-cluster-name",
+		},
+	}
+
+	// Setup the Manager and Controller.  Wrap the Controller Reconcile function so it writes each request to a
+	// channel when it is finished.
+	mgr, err := manager.New(cfg, manager.Options{})
+	g.Expect(err).NotTo(gomega.HaveOccurred())
+	c := mgr.GetClient()
+
+	reconciler := newTestReconciler(mgr, cfnSvcConfig{FailDescribe: true})
+
+	recFn, requests := SetupTestReconcile(reconciler)
+	g.Expect(add(mgr, recFn)).NotTo(gomega.HaveOccurred())
+
+	stopMgr, mgrStopped := StartTestManager(mgr, g)
+
+	defer func() {
+		close(stopMgr)
+		mgrStopped.Wait()
+	}()
+
+	g.Expect(c.Create(context.TODO(), cluster)).NotTo(gomega.HaveOccurred())
+	defer c.Delete(context.TODO(), cluster)
+
+	// Create the ControlPlane object and expect the Reconcile and Deployment to be created
+	err = c.Create(context.TODO(), instance)
```

Comment:
There is no delete for instance.  This is run against a full k8s api, make sure that you either explicitly call delete, or you defer delete.